### DISCUSSION TITLE
vscode plugin and logo suggestion for beve

### DISCUSSION THREAD

**Discussion by meftunca** [beve-org/beve#6] [Ideas]
I am working on 2 different data storage units based on rows and columns. I am using beve and glaze together for this project. I am planning to create a vscode plugin to better see the beve files I am currently creating. 

I am working on 2 different data storage units based on rows and columns. I am using beve and glaze together for this project. I am planning to create a vscode plugin to better see the beve files I am currently creating. I think it can be support for those who may need it in the future :) I am currently waiting for a promise from the project developers that we have a stable beve reader/writer. 

Finally, I am happy to say that we need a logo for beve :)

**Comment by stephenberry** [comment]
A VSCode plug-in to visualize BEVE would be awesome. And, I can confirm that the current specification is stable and there is no intention to break anything. It may be extended, as it has some reserved memory in the header, but we don't intend to break the current spec.

**Comment by stephenberry** [comment]
An svg logo should be made :)

**Comment by meftunca** [comment]
The new plugin is the reason why I can't sleep at this night :). I am working on the plugin and if I can finish it quickly I will share the link here tonight :)

**Reply by stephenberry** [reply]
Haha, neat! Should be really useful.

**Reply by meftunca** [reply]
I can't move forward without seeing what's in front of me :D

**Comment by meftunca** [comment]
![Screenshot 2024-06-23 at 17 27 26](https://github.com/beve-org/beve/assets/19187204/bc70f5f8-9a49-4016-816f-b2ac6a0e2768)
https://github.com/meftunca/beve-vscode

I was able to provide a partial beve->json preview by advancing 30% :D

**Reply by meftunca** [reply]
@stephenberry My temporary new logo suggestion :)  ![image](https://github.com/beve-org/beve/assets/19187204/9e66dad4-227b-4a06-905f-6514c33f5723)

https://www.figma.com/design/G82vKyxe4meIewoWb1ZsBF/BEVE-Logo?node-id=0-1&t=IgW6IDXmnr8acUxe-1

**Reply by meftunca** [reply]
![Screenshot 2024-06-23 at 22 53 46](https://github.com/beve-org/beve/assets/19187204/794a08bc-33fb-49a7-9673-7118b6bc0a94)
I think it looks cute :D

**Reply by meftunca** [reply]
@stephenberry I also list below all the features of the plugin
1 -> Open JSON Editor (Since it is a binary format, vscode does not allow it to open automatically when the file is opened by default, if I can overcome this, the plugin will be completed successfully)
2-> To JSON
3-> From JSON

**Reply by meftunca** [reply]
![image](https://github.com/beve-org/beve/assets/19187204/67a998e2-4ef7-49f3-ac85-6dd65f60cd85)
@stephenberry What do you think of this logo for the file extension?

**Reply by stephenberry** [reply]
I'm excited to check out your plugin! I think the logo should have all the letters `beve`, so I think this logo works. I think a logo like your first `.bv` would also work, but with all the letters: `.beve`. I'd be good with either of these so choose your favorite!

**Comment by meftunca** [comment]
The only thing I'm stuck on at the moment in terms of progress is this:
If you click on the beve file it will tell you that it is not a text file and you will have to "open anyway" and select text format first and then beve format from the file types field in the bottom right corner. Yes this is nonsense and I can't solve it in any way. 

Even if I don't want to, I try to make it work like this by extending the path a little bit.
1. Right click on the file and click on Open as JSON EDITOR
2. Do your editing on the JSON editor (actually a temporary json file)
3. Automatically convert to beve when saved and delete the temporary file if it closes

Please share with me any suggestions

**Reply by stephenberry** [reply]
It sounds like this approach of a temporary JSON file might be a decent solution. I'll look into the plugin and think about possible solutions when I get some time.

**Reply by meftunca** [reply]
Currently, thanks to the JSON Editor feature, if users want to update the beve file 
1. A json file is created in the operating system temporary directory
2. Converts beve to json and saves it in this file
3. If this temporary file is updated, beve is also updated

I think it's really nice, simple and plain :D

**Reply by meftunca** [reply]
As it turned out, it was my first vscode plugin :D 
I really enjoyed it. It meets all my needs now :)

**Reply by stephenberry** [reply]
Neat! I briefly tried it out, but couldn't figure out how to use it properly. It kept giving me errors with the beve file I had open. It said I needed to select a file.

**Reply by meftunca** [reply]
Which version are you using? If you are using the latest version, you can open the editor by right-clicking on the file in the File Explorer section.

**Reply by meftunca** [reply]
"files.associations": {
    "*.beve": "beve"
}

Add userSettings.json

**Reply by stephenberry** [reply]
I got it working out of the box. Really nice! I was having issues because I had told it to view the binary beforehand.

**Reply by meftunca** [reply]
Yes, vscode does not allow us to use binary files as text files. I am waiting for your criticisms, my friend :)

**Comment by stephenberry** [comment]
@meftunca, do you want to move this repository into the https://github.com/beve-org group? No pressure, but I just invited you to the organization so you should be able to add repositories.

**Reply by meftunca** [reply]
Of course my friend. I would be happy :) I will move it tomorrow.

**Reply by meftunca** [reply]
> You don’t have the permission to create public repositories on beve-org
Unfortunately the invitation didn't come, can you send it again?
